# Enter Name here: Ryan McCarthy

Please complete your final project in the space below. Do not forget to explain and interpret the process. You can find the rubric here: https://nathanmichalewicz.org/courses/python/assignments/project-rubric.html

# PROJECT DESCRIPTION

I have chosen to use the "US Collegiate Sports Dataset" for my project. I think this dataset is interesting because I'm a huge fan of the South Carolina Gamecocks football team and the SEC Division. I also play Division I lacrosse. I'd like to learn how much football programs contributed to the overall sports revenue at SEC schools in 2019, whether men's lacrosse programs created more revenue at Division I schools than Division II schools in the years 2015 to 2019, and how many women played lacrosse at Division I schools in 2019.

# VARIABLE ANALYSIS

In the first cell import your libraries and load your data.

Complete one t-test and one ANOVA test.

The t-test and the ANOVA test should not include the same variables and should be related to your research question. Explain why you chose the variables you chose and interpret your results.

In [47]:
# Import libraries and load the dataset

import kagglehub
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import scipy.stats as stats

from statistics import mean, median, mode
from kagglehub import KaggleDatasetAdapter

# Set the debug flag for this segment of code
debug = False

# Download the latest version
path = kagglehub.dataset_download("umerhaddii/us-collegiate-sports-dataset")

# Validate that the file was downloaded
if debug: print(f'Path to dataset files: {path}')

file_path = "sports.csv"

# Import the data into a dataframe
df_sports = kagglehub.load_dataset(
  KaggleDatasetAdapter.PANDAS,
  "umerhaddii/us-collegiate-sports-dataset",
  file_path,
  # https://github.com/Kaggle/kagglehub/blob/main/README.md#kaggledatasetadapterpandas
)

# Print a few rows of the dataset to get a feel for the data
if debug: print(df_sports)

<ipython-input-47-bd2552eb6fe0>:25: DeprecationWarning: load_dataset is deprecated and will be removed in future version.
  df_sports = kagglehub.load_dataset(
/usr/local/lib/python3.11/dist-packages/kagglehub/pandas_datasets.py:91: DtypeWarning: Columns (8) have mixed types. Specify dtype option on import or set low_memory=False.
  result = read_function(


# T-TEST

The following t-test analysis helps answer the question of "how many women played lacrosse at Division I schools in 2019", by comparing that data to women who played lacrosse at Division II schools in 2019.

T-test Question: "Is there a statistically significant difference between women's participation in lacrosse at Division I and Division II schools in 2019?

The t-test will determine if there's a statistically significant difference between the means of the two groups.

Procedure:
- Get the data for the number of women who participated in Division I lacrosse in 2019.

- Get the data for the number of women who participated in Division II lacrosse in 2019.

- Use a t-test to compare the two sets of data.

- Analyze the t-statistic and p-value.

In [48]:
# Set the debug flag for this segment of code

debug = False

# Initialize parameters

div1_categories = ['NCAA Division I-FBS', 'NCAA Division I-FCS']
div2_categories = ['NCAA Division II without football', 'NCAA Division II with football']
sport_lacrosse = 'Lacrosse'
school_year = 2019

# Function to query the dataset and get the number of women who particiated in
# Lacrosse for the given NCAA Categories in 2019

def get_lacrosse_data(param_ncaa_categories):

    # Args:
    # param_ncaa_categories: A list of target NCAA categories.

    # Returns:
    # A list containing the number of women participating in lacrosse for each
    # school in the given NCAA Category in 2019.

    if debug: print(f'\nget_lacrosse_data: ncaa_categories = {param_ncaa_categories}')

    # Query the dataset
    condition = (df_sports['year'] == school_year) & (df_sports['sports'] == sport_lacrosse) & (df_sports['classification_name'].isin(param_ncaa_categories))
    df_results = df_sports.loc[condition, ['year', 'sports', 'institution_name', 'partic_women']].copy()

    if debug: print(f'\ndf_results\n{df_results}')

    # Remove rows where 'partic_women' has blanks and NaN values
    df_results = df_results.dropna(subset=['partic_women'])

    # Get the number of women participants
    partic_women = df_results['partic_women']

    return partic_women

# Function that calculates and outputs the mean, median, and mode of a given list
# of numbers. The mean is the average of all values, the median is the middle value
# when the data is ordered, and the mode is the value that appears most frequently.

def print_stats(param_title, param_numbers):

    # Args:
    # param_title: The name of the list.
    # param_numbers: A list of numbers.

    print(f"\nprint_stats for {param_title}\nmean: {mean(param_numbers)}\nmedian: {median(param_numbers)}\nmode: {mode(param_numbers)}")


# Get the data for the number of women who particiated in Division I Lacrosse in 2019
div1_women = get_lacrosse_data(div1_categories)

if debug: print(f'\ndiv1_women\n{div1_women}')

# Output the mean, median and mode for div1_women
print_stats('Division I Women', div1_women)

# Get the data for the number of women who particiated in Division II Lacrosse in 2019
div2_women = get_lacrosse_data(div2_categories)

if debug: print(f'\ndiv2_women\n{div2_women}')

# Output the mean, median and mode for div2_women
print_stats('Division II Women', div2_women)

# Perform the t-test
t_stat, p_value = stats.ttest_ind(div1_women, div2_women)

# Print the results of the t-test
print(f"\nT-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Check for statistical significance (e.g., at a 5% level)
if p_value < 0.05:
   print("Result: There is a statistically significant difference in the number of women participating in NCAA Division I and Division II lacrosse.")
else:
   print("Result: There is NOT a statistically significant difference in the number of women participating in NCAA Division I and Division II lacrosse.")



print_stats for Division I Women
mean: 33.705882352941174
median: 33.0
mode: 30.0

print_stats for Division II Women
mean: 24.435185185185187
median: 25.0
mode: 26.0

T-statistic: 11.816277319702285
P-value: 1.5389265868448235e-24
Result: There is a statistically significant difference in the number of women participating in NCAA Division I and Division II lacrosse.


T-TEST CONCLUSIONS

According to the course material, a high t-statistic value indicates a strong difference between the means of the two groups being compared. Essentially, it suggests that the observed difference is unlikely to have occurred by chance alone.

 Given the high t-statistic (~ 11.8), I conclude that in 2019, there is a strong difference between the number of women that played NCAA Division I lacrosse and the number of women that played NCAA Division II lacrosse.  The differences in mean values (33.7, 23.4) and mode values (30, 26) for these 2 groups supports this conclusion.

This result helps answer the question of "how many women played lacrosse at Division I schools in 2019" (relative to the number of women that played lacrosse at Division II schools that same year).

# ANOVA TEST

The following analysis helps me answer the question of "How much football programs contributed to the overall sports revenue at SEC schools in 2019" by looking at the amount of money a selected set of SEC Division I schools spent (invested) on football versus other big sports (basketball, soccer, lacrosse and volleyball).

ANOVA Question:
Are there statistically significant differences in the total financial investment (dependent variable) among different collegiate sports (independent variable)?

Procedure:

- Gather expense data for football, basketball, soccer, lacrosse and volleyball at selected NCAA Division I schools.

- Run a one-way ANOVA with "sport" as the independent variable and "expenditure" as the dependent variable.

- Analyze the ANOVA results to see if the f-statistic and p-value are significant.

In [50]:
# Set the debug flag for this segment of code

debug = False

# Create the selected list of SEC Division I schools
school_names = ['The University of Alabama','University of Florida', 'University of Georgia',
                'University of South Carolina-Columbia','The University of Texas at Austin']
# Create the list of sports
sports_names = ['Football', 'Basketball', 'Soccer', 'Lacrosse', 'Volleyball']

# Specify the year
school_year = 2019

# Query the data set
condition = (df_sports['institution_name'].isin(school_names)) & (df_sports['year'] == school_year) & (df_sports['sports'].isin(sports_names))
df_results = df_sports.loc[condition, ['year', 'institution_name', 'sports', 'total_exp_menwomen']].copy()

# Remove rows where 'total_exp_menwomen' has blanks and NaN values
df_results = df_results.dropna(subset=['total_exp_menwomen'])

if debug: print(df_results)

# Prepare input to the ANOVA calculation
exp_football = df_results.loc[df_results['sports'] == sports_names[0], 'total_exp_menwomen']
exp_basketball = df_results.loc[df_results['sports'] == sports_names[1], 'total_exp_menwomen']
exp_soccer = df_results.loc[df_results['sports'] == sports_names[2], 'total_exp_menwomen']
exp_lacrosse = df_results.loc[df_results['sports'] == sports_names[3], 'total_exp_menwomen']
exp_volleyball = df_results.loc[df_results['sports'] == sports_names[4], 'total_exp_menwomen']

if debug:
  print(f'\nTEST-1 exp_football: Length = {len(exp_football)}, Values = {exp_football.values}')
  print(f'\nTEST-2 exp_baseketball: Length = {len(exp_basketball)}, Values = {exp_basketball.values}')
  print(f'\nTEST-3 exp_soccer: Length = {len(exp_soccer)}, Values = {exp_soccer.values}')
  print(f'\nTEST-4 exp_lacrosse: Length = {len(exp_lacrosse)}, Values = {exp_lacrosse.values}')
  print(f'\nTEST-5 exp_volleyball: Length = {len(exp_volleyball)}, Values = {exp_volleyball.values}')

# Output the mean, median and mode for each sport
print_stats('Football', exp_football)
print_stats('Basketball', exp_basketball)
print_stats('Soccer', exp_soccer)
print_stats('Lacrosse', exp_lacrosse)
print_stats('Volleyball', exp_volleyball)

# Calculate the ANOVA values for expenditures across the set of sports
df_anova = stats.f_oneway(exp_football, exp_basketball, exp_soccer, exp_lacrosse, exp_volleyball)

# Outout the results
text_str = f'\nExpenditures for Football, Basketball, Soccer, Lacrosse and Volleyball\nF-statistic: {df_anova.statistic}\np-value: {df_anova.pvalue}'
print(text_str)



print_stats for Football
mean: 43659005.0
median: 39503076.0
mode: 58508853.0

print_stats for Basketball
mean: 14438291.6
median: 13543284.0
mode: 13235476.0

print_stats for Soccer
mean: 2130293.8
median: 1883398.0
mode: 1883398.0

print_stats for Lacrosse
mean: 1648659.0
median: 1648659.0
mode: 1648659.0

print_stats for Volleyball
mean: 2300973.8
median: 1909222.0
mode: 1909222.0

Expenditures for Football, Basketball, Soccer, Lacrosse and Volleyball
F-statistic: 58.15951156367277
p-value: 2.4950602357714594e-09


ANOVA TEST CONCLUSIONS

According to the course material, the high F-statistic (~ 58.16) indicates that there is a statistically significant difference between at least two of the sports.  The extremely small p-value means that the evidence against the null hypothesis (that all group means are equal) is extremely strong.

The ANOVA results indicate that there is a statistically significant difference in the amount of money spent on football, basketball, soccer, lacrosse and volleyball at the selected SEC Division I schools in 2019.

The ANOVA results combined with a comparison of the mean and mode values for  the selected sports at the selected Division I SEC schools indicates that football might have had the highest sports expenditure in 2019, which is also an indicator that football might have made the highest contribution to sports revenue at Division I SEC schools in 2019.  More analyis is required to determine a more accurate answer.

- A key part of this analyis was performed in the previous assignment where I used Pearson Correlation Coefficient analysis to correlate the amount of revenue generated by football with the total amount of revenue generated by all sports at each selected SEC school in 2019.

- The results (0.97 coefficient and 0.00 p-value) indicated a strong correlation between the amount of revenue generated by football and the amount of revenue generated by the overall sports programs at those schools.